cpa/imagetools.py

```diff
@@ -66,22 +70,23 @@ def FetchImage(imKey):
             log_io = wx.GetApp().frame.log_io
         except:
             log_io = True
-        imgs = ir.ReadImages(filenames, log_io)
+        imgs = ir.ReadImages(filenames, log_io, z=z)
         if imgs is None:
             # Loading failed
             return
         cache[imKey] = imgs
-        cachedkeys += [imKey]
+        if imKey not in cachedkeys: #only add the key if it's new
```

Comment:
Unfortunately, without this check I get an error when loading any image after the first: `Traceback (most recent call last):
  File "/Users/rsenft/opt/anaconda3/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/Users/rsenft/Documents/GitHub/CellProfiler/CellProfiler-Analyst/cpa/tilecollection.py", line 185, in run
    new_data = imagetools.FetchTile(obKey, display_whole_image=display_whole_image)
  File "/Users/rsenft/Documents/GitHub/CellProfiler/CellProfiler-Analyst/cpa/imagetools.py", line 33, in FetchTile
    imgs = FetchImage(imKey, z=z)
  File "/Users/rsenft/Documents/GitHub/CellProfiler/CellProfiler-Analyst/cpa/imagetools.py", line 82, in FetchImage
    return cache[imKey]` I implemented that check because it solved this issue but if there's a less expensive way to fix it I'm happy to go with something else.